src/init.cpp

```diff
@@ -1144,6 +1145,10 @@ bool AppInitParameterInteraction(const ArgsManager& args)
         return InitError(Untranslated("peertimeout cannot be configured with a negative value."));
     }
 
+    if (args.GetArg("-pingtimeout", TIMEOUT_INTERVAL) <= 0) {
```

Comment:
Are cases where the response is malformed (e.g. user inputs a string instead of a number) handled elsewhere?

Reply:
Argument handling is in `ArgsManager::GetArg()`. Since the second argument here is an int, the `GetArg(string, int)` version is called:

https://github.com/bitcoin/bitcoin/blob/1e69800d5e60c7805931f2c5b978720c6aee180b/src/util/system.cpp#L473-L477

which will try to convert the string into an int.

This might result in a slightly inaccurate error message, but I think it's fine. It's the same behaviour as for `-peertimeout`:

```
→ ./src/bitcoind --pingtimeout=abc
Error: pingtimeout cannot be configured with a negative value.
```